applications/nrf_desktop/src/modules/failsafe.c

```diff
@@ -15,14 +15,24 @@
 LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_FAILSAFE_LOG_LEVEL);
 
 
-static bool failsafe_check(void)
+static bool failsafe_check(bool *ret)
 {
-	uint32_t mask = NRFX_RESET_REASON_DOG_MASK |
-			NRFX_RESET_REASON_LOCKUP_MASK;
+	int err;
+	uint32_t reas = 0;
+	uint32_t mask = RESET_WATCHDOG |
+			RESET_CPU_LOCKUP;
 
-	uint32_t reas = nrfx_reset_reason_get();
+	err = hwinfo_get_reset_cause(&reas);
+	if (err) {
+		return err;
+	}
+
+	*ret = ((reas & mask) != 0);
+
+	LOG_INF("Reset reason (0x%08X) trigger %s for Failsafe erase operation",
```

Comment:
Can we make it a bit shorter: `"Reset reason (0x%08X), trigger %s`? Log would be prefixed with module name by the logger anyway